Trilinos nightly configuration fails with Kokkos' develop branch

**Describe the bug**
The nightly Trilinos build, Serial backend, with clang/10 began failing to configure with output:
```
CMake Error at kokkos/cmake/kokkos_enable_options.cmake:37 (KOKKOS_ENABLE_OPTION):
  KOKKOS_ENABLE_OPTION Function invoked with incorrect arguments for function
  named: KOKKOS_ENABLE_OPTION
Call Stack (most recent call first):
  kokkos/cmake/kokkos_tribits.cmake:201 (INCLUDE)
  kokkos/CMakeLists.txt:226 (KOKKOS_SETUP_BUILD_ENVIRONMENT)


CMake Error at kokkos/cmake/kokkos_enable_options.cmake:113 (MESSAGE):
  Internal logic error: option 'CUDA_LAMBDA' or device 'CUDA' not recognized.
Call Stack (most recent call first):
  kokkos/cmake/kokkos_enable_options.cmake:123 (CHECK_DEVICE_SPECIFIC_OPTIONS)
  kokkos/cmake/kokkos_tribits.cmake:201 (INCLUDE)
  kokkos/CMakeLists.txt:226 (KOKKOS_SETUP_BUILD_ENVIRONMENT)
```

This was with a Serial build, no configure options involving the CUDA backend (in particular Kokkos_ENABLE_CUDA_LAMBDA) were passed to the cmake configure line


This began after merge of the following commits on May 19:
Only pass one wrapper object in SYCL reductions ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    Explicitly cast to CombinedFunctorReducerType ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    Expand list of kokkos options not to export with cmake ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    Do not append to Kokkos_OPTIONS variables those in the do not export list ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    Drop Kokkos_ENABLE_LAUNCH_COMPILER option ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    Export Kokkos_ENABLE_<OPTION> that are relevant ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    Suppress bogus warning about CUDA_LAMBDA being ON ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    [ci skip] Add nightly ci for spack (#6135) ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))
    OpenMPTarget: Enable Cray compiler for the OpenMPTarget backend. (#5889) ([detail](https://jenkins-son.sandia.gov/job/KokkosEco_Trilinos_Blake_Clang1001_Serial_NoMPI_debug-develop/122/changes#detail))


**Please also include the following items to support reproducing the bug**
**Reproducer** (Blake testbed, SKX arch, clang/10 Serial build):
```bash
# Repos
git clone -b develop https://github.com/trilinos/Trilinos.git
git clone -b develop https://github.com/kokkos/kokkos.git
git clone -b develop https://github.com/kokkos/kokkos-kernels.git
# Symbolic link to your kokkos and kokkos-kernels repos in Trilinos source directory for source override
cd Trilinos
ln -s <path-to-your-repo>/kokkos kokkos
ln -s <path-to-your-repo>/kokkos-kernels kokkos-kernels

cd $HOME
mkdir -p build
cd build

module load cmake/3.25.2 clang/10.0.1 openblas/0.3.13/gcc/10.2.0

cmake \
 -D CMAKE_INSTALL_PREFIX="${TRILINOS_INSTALL_DIR}" \
 -D CMAKE_CXX_STANDARD="17" \
 -D CMAKE_CXX_FLAGS="-g" \
 -D CMAKE_BUILD_TYPE=DEBUG \
 -D BUILD_SHARED_LIBS:BOOL=ON \
\
 -D TPL_ENABLE_MPI=OFF \
\
 -D TPL_ENABLE_BLAS:STRING=ON \
  -D BLAS_LIBRARY_DIRS:FILEPATH=${BLAS_ROOT}/lib \
  -D BLAS_LIBRARY_NAMES:STRING="openblas" \
 -D TPL_ENABLE_LAPACK:STRING=ON \
  -D LAPACK_INCLUDE_DIRS:FILEPATH="${LAPACK_ROOT}/include" \
  -D LAPACK_LIBRARY_DIRS:FILEPATH=${LAPACK_ROOT}/lib \
  -D LAPACK_LIBRARY_NAMES:STRING="openblas" \
\
 -D Trilinos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_EXAMPLES=ON \
 -D Trilinos_ENABLE_COMPLEX=ON \
\
 -D Trilinos_ENABLE_Ifpack2=ON \
  -D Ifpack2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Amesos2=ON \
  -D Amesos2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Kokkos=ON \
 -D Kokkos_ENABLE_SERIAL=ON \
 -D Kokkos_ARCH_SKX=ON \
  -D Kokkos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_KokkosKernels=ON \
  -D KokkosKernels_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Tpetra=ON \
  -D Tpetra_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Sacado=ON \
  -D Sacado_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Stokhos=ON \
  -D Stokhos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Zoltan2=ON \
  -D Zoltan2_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Intrepid2=ON \
  -D Intrepid2_ENABLE_TESTS=OFF \
 -D Trilinos_ENABLE_Belos=ON \
  -D Belos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Anasazi=ON \
  -D Anasazi_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Teuchos=ON \
  -D Teuchos_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_MueLu=ON \
  -D MueLu_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Panzer=ON \
  -D Panzer_ENABLE_TESTS=ON \
 -D Trilinos_ENABLE_Phalanx=ON \
  -D Phalanx_ENABLE_TESTS=ON \
 -D Kokkos_SOURCE_DIR_OVERRIDE:STRING=kokkos \
 -D KokkosKernels_SOURCE_DIR_OVERRIDE:STRING=kokkos-kernels \
$TRILINOS_DIR
```

--- a/cmake/kokkos_enable_options.cmake
+++ b/cmake/kokkos_enable_options.cmake
@@ -34,7 +34,14 @@ KOKKOS_ENABLE_OPTION(CUDA_LDG_INTRINSIC   OFF "Whether to use CUDA LDG intrinsic
 # bogus warning, but also exports the Kokkos_ENABLE_CUDA_LAMBDA variable and
 # sets it to ON. This if-clause is a crutch that delays the refactoring of the
 # way we declare all options until after we get rid of TriBITS.
-KOKKOS_ENABLE_OPTION(CUDA_LAMBDA ${Kokkos_ENABLE_CUDA} "Whether to allow lambda expressions on the device with NVCC **DEPRECATED**")
+IF (Trilinos_ENABLE_Kokkos AND TPL_ENABLE_CUDA)
+   SET(CUDA_LAMBDA_DEFAULT ON)
+ELSEIF (KOKKOS_ENABLE_CUDA)
+   SET(CUDA_LAMBDA_DEFAULT ON)
+ELSE()
+   SET(CUDA_LAMBDA_DEFAULT OFF)
+ENDIF()
+KOKKOS_ENABLE_OPTION(CUDA_LAMBDA ${CUDA_LAMBDA_DEFAULT} "Whether to allow lambda expressions on the device with NVCC **DEPRECATED**")
 
 # As of 08/12/2021 CudaMallocAsync causes issues if UCX is used as MPI communication layer.
 KOKKOS_ENABLE_OPTION(IMPL_CUDA_MALLOC_ASYNC      OFF  "Whether to enable CudaMallocAsync (requires CUDA Toolkit 11.2)")

